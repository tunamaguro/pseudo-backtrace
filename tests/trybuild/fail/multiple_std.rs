use pseudo_backtrace::StackError;

#[derive(Debug, StackError)]
struct MultiStd {
    #[stack_error(std)]
    first: std::io::Error,
    #[stack_error(std)]
    second: std::io::Error,
    #[location]
    location: &'static core::panic::Location<'static>,
}

impl core::fmt::Display for MultiStd {
    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
        "multi".fmt(f)
    }
}
impl core::error::Error for MultiStd {}

fn main() {}
