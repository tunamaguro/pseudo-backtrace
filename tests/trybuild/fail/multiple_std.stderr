error: multiple fields marked with #[stack_error(std)]
 --> tests/trybuild/fail/multiple_std.rs:7:5
  |
7 |     #[stack_error(std)]
  |     ^
